File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Masters/getmasterobject.rb
module Asposediagramjava
  module GetMasterObject
    def initialize()
        # Getting a Master Object by ID
        get_master_object_by_id()

        # Getting a Master Object by Name
        get_master_object_by_name()        
    end

    def get_master_object_by_id() 
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        master_id = 2
        # Get master object by id
        master = diagram.getMasters().getMaster(master_id)

        puts "Master ID : " + master.getID().to_s
        puts "Master Name : " + master.getName().to_s
    end

    def get_master_object_by_name() 
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # Set master name
        master_name = "Background tranquil .2"
        # Get master object by name
        master = diagram.getMasters().getMasterByName(master_name)

        puts "Master ID : " + master.getID().to_s
        puts "Master Name : " + master.getName().to_s
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithMasters/GetMasterObject.php
<?php
namespace Aspose\Diagram\WorkingwithMasters;

use com\aspose\diagram\Diagram as Diagram;
class GetMasterObject{

    public static function run($dataDir=null){
        # Getting a Master Object by ID
        GetMasterObject::get_master_object_by_id($dataDir);

        # Getting a Master Object by Name
        GetMasterObject::get_master_object_by_name($dataDir);
    }

    public static function get_master_object_by_id($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        $master_id = 2;
        # Get master object by id
        $master = $diagram->getMasters()->getMaster($master_id);

        print "Master ID : ".(string)$master->getID().PHP_EOL;
        print "Master Name : ".(string)$master->getName().PHP_EOL;
    }

    public static function get_master_object_by_name($dataDir=null){

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir."drawing.vsd");

        # Set master name
        $master_name = "Background tranquil .2";
        # Get master object by name
        $master = $diagram->getMasters()->getMasterByName($master_name);

        print "Master ID : ".(string)$master->getID().PHP_EOL;
        print "Master Name : ".(string)$master->getName().PHP_EOL;
    }
}